The making of the Diamond Dataset 



In [11]:
# All pip install statements go here. 

# !pip install pandas
# !pip install pillow
# !pip install ImageHash

In [1]:
# All import statements go here 

import pandas as pd
import numpy as np 
from PIL import Image
import imagehash
import os
import requests
from bs4 import BeautifulSoup
from urllib.parse import urljoin


In [22]:
diamond = pd.read_csv(r'/home/levi/Desktop/Work/Diamond Project/Diamonds/Diamonds/data_round.csv')
diamond2 = pd.read_csv(r'/home/levi/Desktop/Work/Diamond Project/Diamonds2/data_round.csv')

In [23]:
diamond.head()

Id  Shape  Weight Clarity Colour Cut Polish Symmetry Fluorescence  \
0  1795561  ROUND     0.3     SI2      K  EX     VG       EX            F   
1  1789678  ROUND     0.3     VS1      L  GD     VG       VG            N   
2  1791701  ROUND     0.3     SI1      K  EX     VG       EX            N   
3  1799570  ROUND     0.3     SI1      L  VG     VG       VG           ST   
4  1774058  ROUND     0.4     VS2    Q-R  EX     EX       EX            N   

     Messurements   Price                                           Data Url  
0  4.32-4.34×2.69  655.38  https://capitalwholesalediamonds.com/product/0...  
1  4.36-4.40×2.56  686.87  https://capitalwholesalediamonds.com/product/0...  
2  4.28-4.31×2.69  692.93  https://capitalwholesalediamonds.com/product/0...  
3  4.27-4.30×2.68  693.42  https://capitalwholesalediamonds.com/product/0...  
4  4.72-4.75×2.91  701.01  https://capitalwholesalediamonds.com/product/0...

In [15]:
diamond.shape

(2141, 12)

There is a chance that the id number in the table corresponds to the name that the jpg file is saved with. <br>
Also there are a few images where the image does not exist and says "There is no preview available for this file.<br>
There are 2141 rows but there are only 1212 round diamond images, also there are plenty of images in which the preview is not available. 

In [24]:
diamond['Shape'].value_counts()

Shape
ROUND    2141
Name: count, dtype: int64

In [34]:
diamond[diamond['Id'] == '2052906']

Empty DataFrame
Columns: [Id, Shape, Weight, Clarity, Colour, Cut, Polish, Symmetry, Fluorescence, Messurements, Price, Data Url]
Index: []

All the repeated images must be deleted from the dataset. They are quite useless. 

In [18]:
print(os.listdir('/home/levi/Desktop/Work/Diamond Project/Diamonds2/images/round'))

['1774818.png', '1743508.jpg', '1655715.jpg', '1653342.jpg', '1655686.jpg', '1604692.jpg', '1786220.jpg', '1783341.jpg', '1775142.jpg', '1752530.jpg', '1792603.png', '1754052.png', '1773169.png', '1799315.jpg', '1796524.jpg', '1662043.png', '1791571.jpg', '1790488.jpg', '1649791.jpg', '1751191.jpg', '1668649.jpg', '1668740.jpg', '1792132.png', '1647553.png', '1789490.png', '1635556.jpg', '1664005.jpg', '1764391.jpg', '1773410.jpg', '1655955.jpg', '1671073.jpg', '1799237.png', '1777102.jpg', '1653574.jpg', '1615786.jpg', '1742847.jpg', '1775022.jpg', '1787267.jpg', '1657323.jpg', '1801890.png', '1663757.jpg', '1781065.jpg', '1649732.jpg', '1762281.jpg', '1762504.jpg', '1783319.jpg', '1758408.jpg', '1665360.jpg', '1657392.jpg', '1787261.jpg', '1802052.png', '1673057.png', '1757757.png', '1657249.jpg', '1662542.jpg', '1792133.png', '1765948.jpg', '1751169.jpg', '1756089.jpg', '1659543.jpg', '1768417.png', '1657110.jpg', '1753717.png', '1756179.jpg', '1627757.jpg', '1748410.png', '1772177.

All the photos that do not contain images of a diamond- the no preview images- are png files, while the rest that actually do contain images of a diamond are jpg files. <br>
Hence we will delete all the images from the dataset are marked as png. 

In [19]:
def delete_png_files(folder_path):
    
    for filename in os.listdir(folder_path):
        if filename.endswith('.png'):
            
            file_path = os.path.join(folder_path, filename)
            

https://capitalwholesalediamonds.com/ <br>

Scrape this website and make a new dataset. 21000 images 

